contracts/TokenTranchePricing.sol

```diff
@@ -48,9 +48,7 @@ contract TokenTranchePricing is PricingStrategy, Ownable {
   /// @param _tranches uint[] tranches Pairs of (start amount, price)
   function TokenTranchePricing(uint[] _tranches) {
     // Need to have tuples, length check
-    if(_tranches.length % 2 == 1 || _tranches.length >= MAX_TRANCHES*2) {
-      throw;
-    }
+    require((_tranches.length % 2 == 0) && (_tranches.length < MAX_TRANCHES*2));
```

Comment:
Since they are dealing with the same data, think better to use same require(). Have changed condition to _tranches.length <= MAX_TRANCHES * 2 similar to EthTranchePricing.sol.